gpii/node_modules/lifecycleManager/src/LifecycleManagerSession.js

```diff
@@ -71,18 +72,90 @@ var gpii = fluid.registerNamespace("gpii");
         }
     });
 
+    gpii.lifecycleManager.session.setAutoSavePrefs = function (existingAutoSavePrefs, preferences, path) {
+        var prefsValue = fluid.get(preferences, path);
+        if (prefsValue) {
+            fluid.set(existingAutoSavePrefs, path, prefsValue);
+        }
+        return existingAutoSavePrefs;
+    };
+
+    /**
+     * Filter the preferences by metadata defined in the preferences set. Only the preferences that are allowed to be
+     * autosaved will be returned.
+     * @param {Object} preferences - The updated preferences.
+     * @param {Object} metadata - The metadata that defined in the prefs set.
+     * @return {Object} - The preferences to be autosaved.
+     */
+    gpii.lifecycleManager.session.filterPrefsByAutoSave = function (preferences, metadata) {
+        var autoSaveScope;
+        fluid.each(metadata, function (meta) {
+            if (meta.type === "autosave") {
+                autoSaveScope = autoSaveScope || {};
+                fluid.each(meta.scope, function (oneScope) {
+                    var oneScopeArray = fluid.makeArray(oneScope);
+                    fluid.set(autoSaveScope, oneScopeArray, true);
+                });
+            }
+        });
+
+        // Preferences will not be auto saved if the "autosave" metadata is not defined
+        if (!autoSaveScope) {
+            return undefined;
+        }
+
+        var prefsForAutoSave = {};
+        fluid.each(autoSaveScope, function (autoSaveLvl1Value, autoSaveLvl1Path) {
+            // handle scoped common terms that are in nested structure:
+            // "http://registry.gpii.net/applications/org.gnome.desktop.a11y.magnifier": {
+            //     "http://registry.gpii.net/common/magnification"
+            // }
+            if (fluid.isObjectSimple(autoSaveLvl1Value)) {
+                fluid.each(autoSaveLvl1Value, function (isAllowed, autoSaveLvl2Path) {
+                    if (isAllowed === true) {
+                        prefsForAutoSave = gpii.lifecycleManager.session.setAutoSavePrefs(
+                            prefsForAutoSave,
+                            preferences,
+                            [autoSaveLvl1Path, autoSaveLvl2Path]
+                        );
+                    }
+                });
+            }
+
+            if (autoSaveLvl1Value === true) {
+                // handle one level common terms such as
+                // "http://registry.gpii.net/common/magnification": 3
+                prefsForAutoSave = gpii.lifecycleManager.session.setAutoSavePrefs(
+                    prefsForAutoSave,
+                    preferences,
+                    [autoSaveLvl1Path]
+                );
+            }
+        });
+
+        return $.isEmptyObject(prefsForAutoSave) ? undefined : prefsForAutoSave;
```

Comment:
Probably cleaner to move this processing into savePreferences since it is at that point that we might be interested in special-casing our actions.

Reply:
OK never mind I see there are two places you make the check, so may as well leave this interpretation here